openai/easyaz.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import typing
+
+if typing.TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+try:
+    from requests import PreparedRequest, Response, Session
+    from requests.adapters import HTTPAdapter
+except ModuleNotFoundError:
+    print("You have to install the `requests` library (pip install requests) in order to use easyaz.requests")
+    exit(-1)
+
+import azure.identity
+import openai
+
+log = logging.getLogger(__name__)
+
+def init(endpoint: str, *, credential=None, api_version='2022-12-01'):
+    openai.api_type = 'azure_ad'
+    openai.api_key = 'dummy'
+    openai.api_base = endpoint
+    openai.api_version = api_version
+
+    if not credential:
+        credential = azure.identity.DefaultAzureCredential()
+
+    session = Session()
+    adapter = AzHttpAdapter(credential=credential, scopes=[ "https://cognitiveservices.azure.com/.default" ])
+    session.mount(endpoint, adapter)
+    openai.requestssession.set(session)
+
+
+class AzHttpAdapter(HTTPAdapter):
```

Comment:
If we want to support more functionality (e.g. auxiliary header), do we want to extend the adapter? Or we want to make those stuff plug-ins so user can choose what they want?